src/test/ui/typeck/issue-104582.stderr

```diff
@@ -20,6 +20,16 @@ help: use angle brackets instead
 LL |     let my_var: String<String?>;
    |                       ~       ~
 
-error: aborting due to 2 previous errors
+error[E0658]: use of unstable library feature 'allocator_api'
+  --> $DIR/issue-104582.rs:2:24
+   |
+LL |     let my_var: String(String?);
+   |                        ^^^^^^^
+   |
+   = note: see issue #32838 <https://github.com/rust-lang/rust/issues/32838> for more information
+   = help: add `#![feature(allocator_api)]` to the crate attributes to enable
+
+error: aborting due to 3 previous errors
```

Comment:
(IIUC) The way that this lexer/parser error is recovered from converts this code to `let my_var: String<String>;`, which now looks like usage of the allocator generic parameter of `String.

Should this test be changed to use a different type (e.g. a `struct` declared in the test) to avoid this?